src/TestLogger/Core/InputSanitizerJson.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Spekt Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Spekt.TestLogger.Core
+{
+    using System.Text;
+
+    public class InputSanitizerJson : IInputSanitizer
+    {
+        private static readonly char[] EscapeTable;
+        private static readonly char[] EscapeCharacters = { '"', '\\', '\b', '\f', '\n', '\r', '\t' };
+
+        static InputSanitizerJson()
+        {
+            EscapeTable = new char[93];
+            EscapeTable['"'] = '"';
+            EscapeTable['\\'] = '\\';
+            EscapeTable['\b'] = 'b';
+            EscapeTable['\f'] = 'f';
+            EscapeTable['\n'] = 'n';
+            EscapeTable['\r'] = 'r';
+            EscapeTable['\t'] = 't';
+        }
+
+        public string Sanitize(string input)
+        {
+            var sb = new StringBuilder();
+
+            if (input is null)
+            {
+                return null;
+            }
+
+            // Happy path if there's nothing to be escaped. IndexOfAny is highly optimized (and unmanaged)
+            if (input.IndexOfAny(EscapeCharacters) == -1)
+            {
+                return input;
+            }
+
+            int safeCharacterCount = 0;
+            char[] charArray = input.ToCharArray();
```

Comment:
It will create a new array so I would avoid it. You can iterate the char on the input string directly.

Reply:
See https://github.com/spekt/testlogger/pull/37#discussion_r1141098585